inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -629,3 +632,50 @@ def load_from_source_data(obj, eng):
         obj.save()
     except KeyError:
         raise ValueError("Can't start/restart workflow as 'source_data' is either missing or corrupted")
+
+
+def restart_workflow(wf, original_workflow, position=[0]):
+    """Restarts workflow
+
+    Args:
+        wf: Workflow to restart
+        original_workflow: Workflow which restarts
+        position: To which position wf should be restarted
+    """
+    wf.callback_pos = position
+    wf.extra_data['source_data']['extra_data']['delay'] = 10
```

Comment:
what's the delay for?

Reply:
It's important to delay workflow restart. It may happen that workflow restarted(R) by previous workflow(P) will fail as workflow "P" still has status "RUNNING" as restarting workflow "R" is actually last step in the workflow "P". Delay happens as first function in first step of workflow "R".

Reply:
but you can't know how long it will take before the other one changes from `RUNNING` to `COMPLETED`. What happens if the other guy is still `RUNNING` after 10s?

Reply:
We have to be reasonable here. 10s is really long time for workflow to change state from `RUNNING` to `COMPLETED` as this is really last part of last step.

Reply:
Is it? `WAIT_FOR_LEGACY_WEBCOLL` step is few steps before `FIND_NEXT_AND_RUN_IF_NECESSARY` step